src/directives/NgGrid.ts

```diff
@@ -1072,11 +1072,11 @@ export class NgGrid implements OnInit, DoCheck, OnDestroy {
 								}
 							}
 
-							if (lowest != itemPos.col) {	//	If the item is not already on this col move it up
+							if (lowest != itemPos.col && this._isWithinBoundsX({lowest, r}, itemDims))) {	//	If the item is not already on this col move it up

```

Comment:
Can you explain the reasoning behind this change to me? We know it's within the bounds due to the fact it's already in this location. Are you finding a situation where the value of `lowest` is greater than `itemPos.col`?